dts/arm/microchip/pic32c/pic32cx_sg/pic32cxsg.dtsi

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2024 Microchip
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <zephyr/dt-bindings/adc/adc.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+
+/ {
+	aliases {
+		adc-0 = &adc0;
+		adc-1 = &adc1;
+
+		port-a = &porta;
+		port-b = &portb;
+		port-c = &portc;
+		port-d = &portd;
+
+		sercom-0 = &sercom0;
+		sercom-1 = &sercom1;
+		sercom-2 = &sercom2;
+		sercom-3 = &sercom3;
+		sercom-4 = &sercom4;
+		sercom-5 = &sercom5;
+		sercom-6 = &sercom6;
+		sercom-7 = &sercom7;
+
+		tc-0 = &tc0;
+		tc-2 = &tc2;
+		tc-4 = &tc4;
+		tc-6 = &tc6;
+
+		tcc-0 = &tcc0;
+		tcc-1 = &tcc1;
+		tcc-2 = &tcc2;
+		tcc-3 = &tcc3;
+		tcc-4 = &tcc4;
+
+		watchdog0 = &wdog;
+	};
+
+	chosen {
+		zephyr,entropy = &trng;
+		zephyr,flash-controller = &nvmctrl;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			device_type = "cpu";
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv7m-mpu";
+				reg = <0xe000ed90 0x40>;
+			};
+		};
+	};
+
+	soc {
+		id: device_id@8061fc {
+			compatible = "atmel,sam0-id";
+			reg = <0x008061FC 0x4>,
+				<0x00806010 0x4>,
+				<0x00806014 0x4>,
+				<0x00806018 0x4>;
+		};
+
+		sram0: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 0x40000>;
+		};
+
+		mclk: mclk@40000800 {
+			compatible = "atmel,sam0-mclk";
+			reg = <0x40000800 0x400>;
+			#clock-cells = <2>;
+		};
+
+		osc32kctrl: osc32kctrl@40001400 {
+			compatible = "atmel,sam0-osc32kctrl";
+			reg = <0x40001400 0x400>;
+			#clock-cells = <0>;
+			#atmel,assigned-clock-cells = <1>;
+		};
+
+		gclk: gclk@40001c00 {
+			compatible = "atmel,sam0-gclk";
+			reg = <0x40001c00 0x400>;
+			#clock-cells = <1>;
+			#atmel,assigned-clock-cells = <1>;
+		};
+
+		wdog: watchdog@40002000 {
+			compatible = "atmel,sam0-watchdog";
+			reg = <0x40002000 13>;
+			interrupts = <10 0>;
+			status = "disabled";
+		};
```

Comment:
Drop all devicetree drivers that are not necessary to run the CPU.
They should be submitted in separated PRs.